botorch/utils/testing.py

```diff
@@ -93,7 +93,7 @@ def assertAllClose(
         )
 
 
-class BaseTestProblemBaseTestCase:
+class BaseTestProblemTestCaseMixIn:
 
     functions: List[BaseTestProblem]
```

Comment:
Pyre is unhappy about the uninitialized attribute. One way to deal with this would be to make `functions` an abstract property. Another way would be to just have `functions: List[BaseTestProblem] = []`. Either way, the easiest thing might be for me to get the type checker passing in a follow-up diff -- what do you think?

Reply:
So the issue with just initializing it as an empty list is that we might end up generating a bunch of noop tests if we're not careful. So abstractproperty is probably a good call here!